File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int count=0;
        int maxi=INT_MIN;
        for(char& c :s){
            if(c =='('){
                count++;
                maxi = max(maxi,count);
            }
            else if(c==')'){
                count--;
            }
            maxi = max(maxi,count);
        }
        return maxi;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        Stack<Character>st = new Stack<>();
        int maxi = 0;
        for(char ch : s.toCharArray()){
            if(ch == '('){
                st.push(ch);
            }
            else if(ch == ')'){
                maxi = Math.max(maxi,st.size());
                st.pop();
            }
        }
        return maxi;
    }
}